src/json.cc

```diff
@@ -481,4 +494,68 @@ std::unique_ptr<Value> Parser::FromString(const std::string& input)
   return FromStream(stream);
 }
 
+class Parser::ParseState {
+ public:
+  ParseState(std::function<void(std::unique_ptr<Value>)> callback)
+      : builder_(callback),
+        handle_(yajl_alloc(&callbacks, NULL, (void*) &builder_)) {
+    yajl_config(handle_, yajl_allow_comments, 1);
+    yajl_config(handle_, yajl_allow_multiple_values, 1);
+    //yajl_config(handle_, yajl_allow_partial_values, 1);
+    //yajl_config(handle_, yajl_allow_trailing_garbage, 1);
+    //yajl_config(handle_, yajl_dont_validate_strings, 1);
+  }
+
+  ~ParseState() {
+    yajl_status stat = yajl_complete_parse(handle_);
+    if (stat != yajl_status_ok) {
+      std::cerr << "Error in yajl_complete_parse" << std::endl;
+      unsigned char* str = yajl_get_error(handle_, 0, nullptr, 0);
+      std::string error_str((const char*)str);
+      yajl_free_error(handle_, str);
+      throw Exception(error_str);
+    }
+    yajl_free(handle_);
+  }
+
+  yajl_handle& handle() { return handle_; }
+
+ private:
+  JSONBuilder builder_;
+  yajl_handle handle_;
+};
+
+Parser::Parser(std::function<void(std::unique_ptr<Value>)> callback)
+    : state_(new ParseState(callback)) {}
+
+Parser::~Parser() {}
+
+std::size_t Parser::ParseStream(std::istream& stream) throw(Exception) {
+  const int kMax = 65536;
+  unsigned char data[kMax];
+  size_t total_bytes_consumed = 0;
+  yajl_handle& handle = state_->handle();
+
+  for (;;) {
```

Comment:
Would `while (!stream.eof()) {` be more concise?

Reply:
Yes, it would; also in line 446. It didn't use to be the case, but after an earlier series of refactorings of that code, the loop ended up being simple enough. Done.